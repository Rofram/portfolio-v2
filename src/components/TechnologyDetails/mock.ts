import { TechnologyDetailsProps } from '.'

const mock: TechnologyDetailsProps = {
  title: 'React.JS',
  description:
    'React JS é uma biblioteca JavaScript para a criação de interfaces de usuário — ou UI (user interface). </br></br> Criado em 2011 pelo time do Facebook, o React surgiu com o objetivo de otimizar a atualização e a sincronização de atividades simultâneas no feed de notícias da rede social, entre eles chat, status, listagem de contatos e outros. </br></br> A princípio, todas essas atividades, chamadas de estados, tinham uma descrição muito complexa. Com o React, esta descrição torna-se mais simples, bem como também é simplificada a conexão entre HTML, CSS e JavaScript e todos os componentes de uma página.',
  color: '#1AA1FF',
  img: {
    src: '/img/techs/react.svg',
    alt: 'React.JS'
  }
}

export default mock
