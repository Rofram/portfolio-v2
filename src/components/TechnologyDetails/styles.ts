import styled, { css } from 'styled-components'

type WrapperProps = {
  color: string
}

export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, color }) => css`
    display: flex;
    width: 60rem;
    border-radius: 20px;
    flex-direction: column;
    padding: ${theme.spacings.medium};
    background-color: ${color};

    > div {
      display: flex;
      justify-content: space-between;
      align-items: center;
    }
  `}
`

export const Title = styled.h2`
  font-size: 4.2rem;
`

export const Description = styled.p`
  font-size: 1.8rem;
  margin-top: 6rem;
`
