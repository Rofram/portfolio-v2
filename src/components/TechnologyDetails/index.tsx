import * as S from './styles'

const TechnologyDetails = () => (
  <S.Wrapper>
    <h1>TechnologyDetails</h1>
  </S.Wrapper>
)

export default TechnologyDetails
