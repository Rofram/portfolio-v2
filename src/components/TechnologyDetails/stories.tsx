import { Story, Meta } from '@storybook/react'
import TechnologyDetails from '.'

export default {
  title: 'TechnologyDetails',
  component: TechnologyDetails
} as Meta

export const Default: Story = () => <TechnologyDetails />
