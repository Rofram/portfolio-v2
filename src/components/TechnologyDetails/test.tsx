import { render, screen } from '@testing-library/react'

import TechnologyDetails from '.'

describe('<TechnologyDetails />', () => {
  it('should render the heading', () => {
    const { container } = render(<TechnologyDetails />)

    expect(
      screen.getByRole('heading', { name: /TechnologyDetails/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
