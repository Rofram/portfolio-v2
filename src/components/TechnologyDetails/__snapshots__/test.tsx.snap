// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TechnologyDetails /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    TechnologyDetails
  </h1>
</main>
`
